sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -640,6 +648,19 @@ boolean bucketAccessible(GcsPath path, BackOff backoff, Sleeper sleeper) throws
     }
   }
 
+  /**
+   * Checks whether the GCS bucket exists. Throws exception if the bucket is inaccessible due to
+   * permissions or does not exist.
+   */
+  @VisibleForTesting
+  void verifyBucketAccessible(GcsPath path, BackOff backoff, Sleeper sleeper) throws IOException {
+    try {
+      getBucket(path, backoff, sleeper);
+    } catch (AccessDeniedException | FileNotFoundException e) {
+      throw e;
```

Comment:
What does rethrowing get us here? Won't this exception be thrown regardless (whether its an AccessDeniedException/FileNotFoundException or not)? (That seems like what we'd want as well)

Reply:
Good catch! Removed the try-catch block.